src/doc/reference.md

```diff
@@ -2041,6 +2050,8 @@ type int8_t = i8;
   item](#language-items) for more details.
 - `test` - indicates that this function is a test function, to only be compiled
   in case of `--test`.
+- `should_fail` - indicates that this test function should panic, inverting the success condition.
+- `cold` - The function is unlikely to be executed, so optimize it differently.
```

Comment:
"so optimize it and calls to it", maybe? (IME, the main use is making the compiler rearrange branches to preference the ones that don't have `#[cold]` functions.)